src/wasm-type.h

```diff
@@ -251,17 +261,82 @@ struct ResultType {
   std::string toString() const;
 };
 
+class HeapType {
```

Comment:
I think it would be good to add a comment explaining what a HeapType is, and why it's separate from Type.